src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncMainTests.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
+{
+    public class CodeGenAsyncMainTests : EmitMetadataTestBase
+    {
+        [Fact]
+        public void AsyncEmitMainOfIntTest()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+
+class Program {
+    static async Task<int> Main() {
+        Console.Write(""hello "");
+        await Task.Factory.StartNew(() => 5);
+        Console.Write(""async main"");
+        return 10;
+    }
+}";
+            var c = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7_1));
+            var verifier = CompileAndVerify(c, expectedOutput: "hello async main");
+            Assert.Equal(verifier.LastErrorCode, 10);
```

Comment:
>Assert.Equal(verifier.LastErrorCode, 10); [](start = 12, length = 41)

Consider simply passing expected error code to CompileAndVerify and removing the LastErrorCode property.